service.upnext/resources/lib/script.py

```diff
@@ -0,0 +1,110 @@
+        self.setProperty('clearlogo', 'https://fanart.tv/fanart/tv/121361/hdtvlogo/game-of-thrones-504c49ed16f70.png')
+        self.setProperty('fanart', 'https://fanart.tv/fanart/tv/121361/showbackground/game-of-thrones-4fd5fa8ed5e1b.jpg')
+        self.setProperty('landscape', 'https://fanart.tv/detailpreview/fanart/tv/121361/tvthumb/game-of-thrones-4f78ce73d617c.jpg')
+        self.setProperty('poster', 'https://fanart.tv/fanart/tv/121361/tvposter/game-of-thrones-521441fd9b45b.jpg')
+        self.setProperty('thumb', 'https://fanart.tv/fanart/tv/121361/showbackground/game-of-thrones-556979e5eda6b.jpg')
+
+        self.setProperty('episode', '4')
+        self.setProperty('playcount', '1')
+        self.setProperty('plot', 'Lord Baelish arrives at Renly\'s camp just before he faces off against Stannis. '
+                                 'Daenerys and her company are welcomed into the city of Qarth. Arya, Gendry, and '
+                                 'Hot Pie find themselves imprisoned at Harrenhal.')
+        self.setProperty('rating', '8.9')
+        self.setProperty('season', '2')
+        self.setProperty('seasonepisode', '2x4')
+        self.setProperty('title', 'Garden of Bones')
+        self.setProperty('tvshowtitle', 'Game of Thrones')
+        self.setProperty('year', '2012')
+        self.setProperty('runtime', '50')
+
+    def prepare_progress_control(self):
+        try:
+            self.progress_control = self.getControl(3014)
+        except RuntimeError:
+            return
+        self.progress_control.setPercent(100.0)  # pylint: disable=no-member,useless-suppression
+
+    def update_progress_control(self, timeout, wait):
+        if self.progress_control is None:
+            return
+        self.current_progress_percent -= 100 * wait / timeout
+        self.progress_control.setPercent(self.current_progress_percent)  # pylint: disable=no-member,useless-suppression
+        self.setProperty('remaining', from_unicode('%02d' % ceil((timeout / 1000) * (self.current_progress_percent / 100))))
+        self.setProperty('endtime', from_unicode(localize_time(datetime.now() + timedelta(seconds=50 * 60))))
+
+    def onFocus(self, controlId):  # pylint: disable=invalid-name
+        pass
+
+    def doAction(self):  # pylint: disable=invalid-name
+        pass
+
+    def closeDialog(self):  # pylint: disable=invalid-name
+        self.close()
+
+    def onClick(self, controlId):  # pylint: disable=invalid-name,unused-argument
+        self.close()
+
+    def onAction(self, action):  # pylint: disable=invalid-name
+        if action == self.ACTION_PLAYER_STOP:
+            self.close()
+        elif action == self.ACTION_NAV_BACK:
+            self.close()
+
+
+def test_popup(window):
+    popup = TestPopup(window, addon_path(), 'default', '1080i')
+    popup.show()
+    step = 0
+    wait = 100
+    timeout = 10000
```

Comment:
not sure if this is used but an implementation with waitForAbort and abortRequested() is advised

Reply:
Only in testing, but will have a look or maybe exclude

Reply:
This is a countdown counter for testing pop-up window for theme developers.
I don't see how we would implement this using `waitForAbort` and `abortRequested()`, or wh that is relevant here.

Reply:
We are using `xbmc.sleep()` underneath, is that a problem as well?

Reply:
So I don't think this is a problem, this is broken up into smaller pieces, totalling up to 10 seconds (the time we want the example pop-up to appear on screen).

Reply:
If only used in testing there's no issue with it. There's also no issue if the big timeout chunk is split into smaller wait times as long as the request for abortion is checked in-between. This is usually not requested just for the sake of it, but as a way to avoid plugins/scripts hanging kodi on exit